src/policy_artifacthub.rs

```diff
@@ -0,0 +1,264 @@
+#[serde(rename_all = "camelCase")]
+pub struct ArtifactHubPkg {
+    pub version: Version,
+    pub name: String,
+    pub display_name: String,
+    #[serde(with = "time::serde::rfc3339")]
+    pub created_at: OffsetDateTime,
+    pub description: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub license: Option<String>, // TODO spdx
+    #[serde(rename = "homeURL", skip_serializing_if = "Option::is_none")]
+    pub home_url: Option<Url>, // https://github.com/kubewarden/verify-image-signatures
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub containers_images: Option<Vec<ContainerImage>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub keywords: Option<Vec<String>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub links: Option<Vec<Link>>,
+
+    // #[serde(skip_serializing_if = "Option::is_none")]
+    // pub maintainers: Option<Vec<Maintainer>>,
+    pub provider: Provider, // TODO we can't hardcode it
+
+    pub recommendations: Vec<Recommendation>,
+    pub annotations: HashSet<Annotation>,
+
+    pub readme: Option<String>,
+}
+
+impl Default for ArtifactHubPkg {
+    fn default() -> Self {
+        let now = OffsetDateTime::now_utc();
+        Self {
+            created_at: now,
+            version: Version::new(0,0,0),
+            name: Default::default(),
+            display_name: Default::default(),
+            description: Default::default(),
+            license: Default::default(),
+            home_url: Default::default(),
+            containers_images: Default::default(),
+            keywords: Default::default(),
+            links: Default::default(),
+            provider: Default::default(),
+            recommendations: Default::default(),
+            annotations: Default::default(),
+            readme: Default::default(),
+        }
+    }
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct ContainerImage {
+    pub name: String,  // always "policy"
+    pub image: String, // not URL, e.g: ghcr.io/kubewarden/policies/verify-image-signatures:v0.2.1
+}
+
+// Examples:
+// - name: policy
+//   url: https://github.com/kubewarden/verify-image-signatures/releases/download/v0.2.1/policy.wasm
+// - name: source
+//   url: https://github.com/kubewarden/verify-image-signatures
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct Link {
+    pub name: String, // either "policy", or "source"
+    pub url: Url,
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct Recommendation {
+    #[serde(default = "default_recommendation")]
+    pub url: Url,
+}
+
+fn default_recommendation() -> Url {
+    Url::parse("https://artifacthub.io/packages/helm/kubewarden/kubewarden-controller").unwrap()
+}
+
+// #[derive(Default, Deserialize, Serialize, Debug, Clone)]
+// #[serde(rename_all = "camelCase")]
+// pub struct Maintainer {
```

Comment:
This should conform `address-list` following https://www.rfc-editor.org/rfc/rfc2822#section-3.
Maybe implementable with https://docs.rs/mail-parser/latest/mail_parser/struct.Addr.html, but doubt it can be deserialized without writing a deserialize.

Reply:
Why do we need a custom deserializer?

What I would do is:

* Look for the `io.kubewarden.policy.author` field inside of `metadata.yml`
* Split it by `,`
* Iterate over each entry and try to parse that with the crate you linked
* We should just consider one email format: `Homer Simpson <homer@simpsons.com>`
* If parsing fails, exit with an error and tell what is the email format we support

Reply:
Done it by manually parsing name and email.